### PR TITLE
[Mobile] - Refactor BlockListItem

### LINKED ISSUE #4081: Unselected Column placeholder is NOT stretched to content in nested structure
**Describe the bug**

**To Reproduce**
Follow the steps in [columns/tc017](https://github.com/wordpress-mobile/test-cases/blob/trunk/test-cases/gutenberg/columns.md#tc017) or the following steps:
1. Add a columns block with two columns
2. Click on the first column and add a columns block with two columns
3. Rotate the device
4. **Notice** that the empty column placeholder is not stretched accordingly

**Expected behavior**
The empty Column placeholder to be stretched accordingly

**Screenshots**

|Expected (`18.1`)|Actual (`18.2+`)|
|---|---|
|||

**Smartphone (please complete the following information):**
 - Device: iPhone SE 2020, Google Pixel 5
 - OS: iOS 14.7.1, Android 12
 - Version WordPress iOS/Android `18.2`

### LINKED ISSUE #5615: Refactor BlockListItem component to use custom hook and functional component
The `BlockListItem` component is currently using a class component and the `ReadableContentView` component to handle the max width of the canvas and provide alignment styles. We want to refactor this component to use a custom hook (depending on https://github.com/wordpress-mobile/gutenberg-mobile/issues/5614) and a functional component instead.

**The proposed changes are**:

- Replace the usage of `ReadableContentView` with a custom hook that will handle the max width of the canvas and provide alignment styles. The hook will return just the styles and should leverage the useWindowDimensions hook.
- Calculate the `getMarginHorizontal` in the same custom hook that will be created from `ReadableContentView` to improve performance, then we would just have one `View` that will have all of this data.
- Refactor the `BlockListItem` component to be a functional component instead of a class component.

### PR DESCRIPTION
**Pull request by geriux**
Closes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5615
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4081

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5660

## What?
This PR refactors the `BlockListItem` component by updating the code into a functional component, using the new [useEditorWrapperStyles](https://github.com/WordPress/gutenberg/pull/49616) hook.

## Why?

To keep the code updated with the latest code practices and to simplify its logic.

## How?

- It refactors the component into two functional components: `BlockListItemContent`, and `BlockListItem`
- Removes the logic related to margins and alignments and now uses the [useEditorWrapperStyles](https://github.com/WordPress/gutenberg/pull/49616) hook
- Adds comments to all props and avoids passing spread props from its parent to the `Block` component, this way we know which props are being received and passed down easily
- Simplifies some selectors and if conditions

## Testing Instructions

CI checks should pass for both Gutenberg and Gutenberg mobile.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

**Review by dcalhoun — COMMENTED** [review]
This looks good so far. 👏🏻 I have not performed any manual tests, only reviewed the code itself. 

> Fixes wordpress-mobile/gutenberg-mobile#4081

🎉

**Comment by geriux** [comment]
This branch has been updated with the feature branch, it also includes an extra commit https://github.com/WordPress/gutenberg/pull/49765/commits/869ccb76d6a61e4b50e226c622d3b9e4cbb2f01c that adds two missing props.

**Comment by geriux** [comment]
Hey @dcalhoun 👋 this PR is now targeting `trunk` let me know if you can take a look. Thanks!

File: packages/block-editor/src/components/block-list/block-list-item.native.js [hunk 1]
```diff
@@ -1,237 +1,198 @@
...
+ * @param {number}   props.tileCount                    The total number of tiles in the grid.
+ * @param {number}   props.tileIndex                    The index of the current tile in the grid.
+ * @param {number}   props.blockWidth                   The width of the block.
+ * @param {string}   props.clientId                     The block client ID.
+ * @param {string}   props.contentResizeMode            The content resize mode, e.g "stretch".
+ * @param {Object}   props.contentStyle                 Styles for the block content
+ * @param {boolean}  props.isStackedHorizontally        Whether the block is stacked horizontally.
+ * @param {number}   props.marginHorizontal             The horizontal margin.
+ * @param {number}   props.marginVertical               The vertical margin.
+ * @param {Function} props.onAddBlock                   On Add block callback.
```
Comment (dcalhoun) [inline packages/block-editor/src/components/block-list/block-list-item.native.js]: Suggestion to match casing of other descriptions. 
```suggestion
 * @param {Function} props.onAddBlock                   On add block callback.
```

File: packages/block-editor/src/components/block-list/block-list-item.native.js [hunk 2]
```diff
@@ -1,237 +1,198 @@
...
+ *
+ * @param {Object}   props                              Component properties.
+ * @param {number}   props.blockWidth                   The width of the block.
+ * @param {string}   props.clientId                     The block client ID.
+ * @param {string}   props.contentResizeMode            The content resize mode, e.g "stretch".
+ * @param {Object}   props.contentStyle                 Styles for the block content
+ * @param {boolean}  props.isStackedHorizontally        Whether the block is stacked horizontally.
+ * @param {number}   props.marginHorizontal             The horizontal margin.
+ * @param {number}   props.marginVertical               The vertical margin.
+ * @param {Function} props.onAddBlock                   On Add block callback.
```
Comment (dcalhoun) [inline packages/block-editor/src/components/block-list/block-list-item.native.js]: Suggestion to match casing of other descriptions. 
```suggestion
 * @param {Function} props.onAddBlock                   On add block callback.
```

**Review by dcalhoun — APPROVED** [review]
LGTM. 🚀 

I did a bit of manual testing on an iPhone 14 Pro simulator, iPad Pro simulator, and Pixel 3a emulator. I did not encounter any issues.

File: packages/block-editor/src/components/block-list/block-list-item.native.js (hunk 1 above)
Comment (geriux) [reply packages/block-editor/src/components/block-list/block-list-item.native.js]: Sure! Updated in https://github.com/WordPress/gutenberg/pull/49765/commits/4d3b846d04902a07e3066415dde5da04b7bf4e26

File: packages/block-editor/src/components/block-list/block-list-item.native.js (hunk 2 above)
Comment (geriux) [reply packages/block-editor/src/components/block-list/block-list-item.native.js]: Updated in https://github.com/WordPress/gutenberg/pull/49765/commits/4d3b846d04902a07e3066415dde5da04b7bf4e26

**Pull request merged**